nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/HealthCheckServlet.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.standard.servlets;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.Path;
+import java.io.IOException;
+
+@Path("/healthcheck")
+public class HealthCheckServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        response.setStatus(HttpServletResponse.SC_OK);
```

Comment:
Some load balancers cannot accept anything else other than a 200 response I believe, no?

Reply:
That's a good question.  There are many load balancing products and services available, here are several that support a configurable range of HTTP response codes to indicate healthy status:

- Apache HTTP Server mod_proxy_hcheck
- Nginx
- Amazon Elastic Load Balancing
- F5 BIG-IP

It would be helpful to know if there are specific products that are limited to HTTP 200 responses.  Perhaps making the success response code a configurable property would be the most flexible option.

Reply:
That is right, 204 would be more precise. Unfortunately AWS Classic Load Balancer expects 200 and does not accept 204. I think it is safe to use 200 because that is accepted by all LBs. However, I'll add some static content to the response body because it can cause confusion that 200 is returned with an empty response body.

Reply:
@pgyori Thanks for the explanation, I found the AWS documentation for reference:

https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/ts-elb-healthcheck.html

Using 200 and returning some static content sounds good.